app/components/Settings/Scheduler/previewModal.jsx

```diff
@@ -0,0 +1,248 @@
+import { FrequencySelector } from './createModal.jsx';
+import { Row, Column } from '../../layout.jsx';
+import SQL from './sql.jsx';
+import { plotlyUrl } from '../../../utils/utils.js';
+import { mapDialect } from './util.js';
+
+const NO_OP = () => {};
+
+const rowStyle = {
+    justifyContent: 'flex-start',
+    borderBottom: '1px solid rgba(0, 0, 0, 0.05)',
+    padding: '16px 0px'
+};
+const boxStyle = { boxSizing: 'border-box', width: '50%' };
+
+export class PreviewModal extends Component {
+    static defaultProps = {
+        onSave: NO_OP,
+        onDelete: NO_OP
+    };
+    static propTypes = {
+        query: PropTypes.object,
+        onSave: PropTypes.func,
+        onDelete: PropTypes.func,
+        onClickAway: PropTypes.func,
+        dialect: PropTypes.string
+    };
+    constructor(props) {
+        super(props);
+        this.state = {
+            editing: false,
+            code: props.query && props.query.query,
+            refreshInterval: props.query && props.query.refreshInterval,
+            confirmedDelete: false
+        };
+        this.toggleEditing = this.toggleEditing.bind(this);
+        this.updateCode = this.updateCode.bind(this);
+        this.updateRefreshInterval = this.updateRefreshInterval.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+        this.onDelete = this.onDelete.bind(this);
+        this.close = this.close.bind(this);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            this.props.query &&
+            this.props.query.query !==
+                (prevProps.query && prevProps.query.query)
+        ) {
+            // eslint-disable-next-line
+            this.setState({
+                code: this.props.query.query,
+                refreshInterval: this.props.query.refreshInterval
+            });
+        }
+    }
+
+    updateCode(editor, meta, code) {
+        this.setState({ code });
+    }
+
+    updateRefreshInterval({ value: refreshInterval }) {
+        this.setState({ refreshInterval });
+    }
+
+    toggleEditing() {
+        this.setState(({ editing }) => ({ editing: !editing }));
+    }
+
+    onSubmit() {
+        if (this.state.editing) {
+            const { connectionId, fid, requestor, uids } = this.props.query;
+            const { code: query, refreshInterval } = this.state;
+            this.props.onSave({
+                connectionId,
+                fid,
+                requestor,
+                uids,
+                query,
+                refreshInterval
+            });
+            this.setState({ editing: false, confirmedDelete: false });
+        } else {
+            this.toggleEditing();
+        }
+    }
+
+    onDelete() {
+        if (this.state.confirmedDelete) {
+            this.setState({ confirmedDelete: false }, () => {
+                this.props.onDelete(this.props.query.fid);
+            });
+        } else {
+            this.setState({ confirmedDelete: true });
+        }
+    }
+
+    close() {
```

Comment:
At the moment, the modal window closes as soon as the user clicks a button. With this behaviour, users can't tell whether the request is still being processed or it has failed.

Ideally, we want to tell the user whether the request failed and if it did fail we don't want to close the modal window.